public/tier1/memblockallocator.h

```diff
@@ -0,0 +1,74 @@
+//===== Copyright 1996-2005, Valve Corporation, All rights reserved. ======//
+//
+// Purpose: 
+//
+// $Workfile:     $
+// $Date:         $
+//
+//-----------------------------------------------------------------------------
+// $Log: $
+//
+// $NoKeywords: $
+//===========================================================================//
+
+#ifndef MEMBLOCKALLOCATOR_H
+#define MEMBLOCKALLOCATOR_H
+
+#ifdef _WIN32
+#pragma once
+#endif
+
+#include "tier1/utlvector.h"
+
+typedef unsigned int MemBlockHandle_t;
+
+#define MEMBLOCKHANDLE_INVALID ((MemBlockHandle_t)~0)
+
+class CUtlMemoryBlockAllocator
```

Comment:
Where did you find this class definition? I'm aware that some methods from here are exported by tier0.dll, my question is where/how did you come up with the structure of that class?